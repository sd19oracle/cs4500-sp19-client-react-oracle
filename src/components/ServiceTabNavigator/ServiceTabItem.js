import React from 'react'

const ServiceTabItem = ({services}) =>
    <div className="row">
        {
            services.map(service =>
                <div key={service.id}
                     className="card col-4 no-border">
                    <img src="https://picsum.photos/300/200" className="card-img-top" alt="..."/>
                    <div className="card-body">
                        <h5 className="card-title">
                            <a href="/providers">{service.title}</a>
                        </h5>
                        {/*<p className="card-text">Some quick example text to build on the card title and make up the bulk of the card's content.</p>*/}
                        {/*<a href="#" className="btn btn-primary">Go somewhere</a>*/}
                    </div>
                </div>
            )
        }
    </div>

export default ServiceTabItem