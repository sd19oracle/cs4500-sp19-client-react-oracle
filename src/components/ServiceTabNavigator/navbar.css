.nav-link {
    color: rgb(18, 126, 251);
}

.nav-link:hover {
    cursor: pointer;
}