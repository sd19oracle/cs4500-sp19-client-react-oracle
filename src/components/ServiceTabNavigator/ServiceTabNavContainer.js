import React from 'react'
import ServiceTabNavigator from './ServiceTabNavigator'
import ServiceService from '../../services/ServiceService'
import serviceCategories from '../../data/popular-service-categories.mock'
import ServiceTabItem from "./ServiceTabItem";

class ServiceTabNavContainer extends React.Component {
    constructor(props) {
        super(props);
        this.serviceService = ServiceService.getInstance();
        this.state = {
            popularServiceCategories: [],
            activeCategoryServices: []
        };
        this.switchCategory = this.switchCategory.bind(this);
    }

    componentDidMount() {
        for (let i in serviceCategories) {
            this.serviceService
                .findPopularServicesByCategory(serviceCategories[i].id, 6)
                .then(services => {
                    let newPopularServices = this.state.popularServiceCategories;
                    newPopularServices.push(
                        {
                            "id": serviceCategories[i].id,
                            "category_name": serviceCategories[i].name,
                            "services": services
                        });
                    this.setState({
                        popularServiceCategories: newPopularServices,
                        activeCategoryServices: this.state.popularServiceCategories[0].services
                    });
                })
        };
    }

    switchCategory(event) {
        for (let i in this.state.popularServiceCategories) {
            if (this.state.popularServiceCategories[i].category_name === event.target.innerText) {
                this.setState({
                    activeCategoryServices: this.state.popularServiceCategories[i].services
                })
            }
        }
    }

    render() {
        return (
            <div>
                <ServiceTabNavigator services={this.state.popularServiceCategories}
                    switchCategory={this.switchCategory}>
                </ServiceTabNavigator>
                <ServiceTabItem services={this.state.activeCategoryServices }/>
            </div>
        )

    }
}

export default ServiceTabNavContainer