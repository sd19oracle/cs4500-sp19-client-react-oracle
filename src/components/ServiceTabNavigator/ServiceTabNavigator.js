import React from 'react'
import ServiceTabItem from './ServiceTabItem'
import {BrowserRouter as Router, Route, Link} from 'react-router-dom'

const ServiceTabNavigator = ({serviceCategories}) =>
    <Router>
        <div>
            <ul className="nav nav-tabs">
                {
                    serviceCategories.map(serviceCategory =>
                        <li key={serviceCategory.id} className="nav-item">
                            <Link to={serviceCategory.title} className="nav-link" href="#">
                                {serviceCategory.title}
                            </Link>
                        </li>
                    )
                }
            </ul>
            <br/>
            <Route path="/Home Improvements" render={() =>
                <ServiceTabItem services={serviceCategories[0].services.splice(0, 6)}/>}/>
            <Route path="/Pets" render={() =>
                <ServiceTabItem services={serviceCategories[1].services.splice(0, 6)}/>}/>
        </div>
    </Router>

export default ServiceTabNavigator