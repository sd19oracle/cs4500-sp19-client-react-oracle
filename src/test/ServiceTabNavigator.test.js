import React from 'react'
import TestRenderer from 'react-test-renderer';
import ServiceTabNavigator from '../components/ServiceTabNavigator/ServiceTabNavigator'
import popularServiceCategories from '../data/popular-service-categories.mock'

// Snapshot testing
test('basic test', () => {
    const testRenderer = TestRenderer.create(
        <ServiceTabNavigator services={popularServiceCategories}/>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
});

// snapshot testing
test('empty service category test', () => {
    const testRenderer = TestRenderer.create(
        <ServiceTabNavigator services={[]}/>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
});

// DOM testings
test('test popular services length', () => {
    const testRenderer = TestRenderer.create(
        <ServiceTabNavigator services={popularServiceCategories}/>
    );

    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

    const testInstance = testRenderer.root;
    const categoryLink = testInstance.findAllByProps({className: 'nav-link'});
    expect(categoryLink.length).toBe(10);
});

test('test nav tab length', () => {
    const testRenderer = TestRenderer.create(
        <ServiceTabNavigator services={popularServiceCategories}/>
    );

    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

    const testInstance = testRenderer.root;
    const navBarDom = testInstance.findAllByProps({className: 'nav nav-tabs'});
    expect(navBarDom.length).toBe(1);
});

