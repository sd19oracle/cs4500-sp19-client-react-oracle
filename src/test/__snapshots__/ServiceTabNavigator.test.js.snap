// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic test 1`] = `
<div>
  <ul
    className="nav nav-tabs"
  >
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Home Improvements
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Pets
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Wellness
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Business
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Events
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Crafts
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Design and Web
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Legal
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Personal
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Photography
      </span>
    </li>
  </ul>
  <br />
</div>
`;

exports[`empty service category test 1`] = `
<div>
  <ul
    className="nav nav-tabs"
  />
  <br />
</div>
`;

exports[`test nav tab length 1`] = `
<div>
  <ul
    className="nav nav-tabs"
  >
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Home Improvements
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Pets
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Wellness
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Business
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Events
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Crafts
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Design and Web
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Legal
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Personal
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Photography
      </span>
    </li>
  </ul>
  <br />
</div>
`;

exports[`test popular services length 1`] = `
<div>
  <ul
    className="nav nav-tabs"
  >
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Home Improvements
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Pets
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Wellness
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Business
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Events
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Crafts
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Design and Web
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Legal
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Personal
      </span>
    </li>
    <li
      className="nav-item"
    >
      <span
        className="nav-link"
      >
        Photography
      </span>
    </li>
  </ul>
  <br />
</div>
`;
